pkg/collector/corechecks/windows_event_log/check.go

```diff
@@ -0,0 +1,294 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build windows
+
+// Package evtlog defines a check that reads the Windows Event Log and submits Events
+package evtlog
+
+import (
+	"fmt"
+	"regexp"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	agentCheck "github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	agentEvent "github.com/DataDog/datadog-agent/pkg/metrics/event"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/api"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/api/windows"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/session"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/subscription"
+
+	"golang.org/x/sys/windows"
+)
+
+const checkName = "win32_event_log"
```

Comment:
We want to use "windows event log" where we can, since that's the proper Windows name. We've even updated the title on the docs page. But we have to keep the check name for compatibility, right?